src/gpu/multiexp.rs

```diff
@@ -287,44 +292,64 @@ where
 
         let chunk_size = ((n as f64) / (num_devices as f64)).ceil() as usize;
 
-        crate::multicore::THREAD_POOL.install(|| {
-            use rayon::prelude::*;
-
-            let mut acc = <G as CurveAffine>::Projective::zero();
+        let mut acc = <G as CurveAffine>::Projective::zero();
 
-            let results = if n > 0 {
+        let results =
+            if n > 0 {
                 bases
-                    .par_chunks(chunk_size)
-                    .zip(exps.par_chunks(chunk_size))
-                    .zip(self.kernels.par_iter_mut())
-                    .map(|((bases, exps), kern)| -> Result<<G as CurveAffine>::Projective, GPUError> {
-                        let mut acc = <G as CurveAffine>::Projective::zero();
-                        for (bases, exps) in bases.chunks(kern.n).zip(exps.chunks(kern.n)) {
-                            let result = kern.multiexp(bases, exps, bases.len())?;
-                            acc.add_assign(&result);
-                        }
-
-                        Ok(acc)
-                    })
-                    .collect::<Vec<_>>()
+                .par_chunks(chunk_size)
+                .zip(exps.par_chunks(chunk_size))
+                .zip(self.kernels.par_iter_mut())
+                .map(|((bases, exps), kern)| -> Result<<G as CurveAffine>::Projective, GPUError> {
+                    let mut acc = <G as CurveAffine>::Projective::zero();
+                    for (bases, exps) in bases.chunks(kern.n).zip(exps.chunks(kern.n)) {
+                        let result = kern.multiexp(bases, exps, bases.len())?;
+                        acc.add_assign(&result);
+                    }
+
+                    Ok(acc)
+                })
+                .collect::<Vec<_>>()
             } else {
                 Vec::new()
             };
 
-            let cpu_acc = cpu_multiexp(
-                &pool,
-                (Arc::new(cpu_bases.to_vec()), 0),
-                FullDensity,
-                Arc::new(cpu_exps.to_vec()),
-                &mut None,
-            );
+        let cpu_acc = cpu_multiexp(
+            &pool,
+            (Arc::new(cpu_bases.to_vec()), 0),
+            FullDensity,
+            Arc::new(cpu_exps.to_vec()),
+            &mut None,
+        );
 
-            for r in results {
-                acc.add_assign(&r?);
-            }
+        for r in results {
+            acc.add_assign(&r.unwrap());
```

Comment:
Is that unwrap save? If yes I think a comment on why would be great.

I also tend to use `except()` instead of `unwrap()`, sometimes you can even put the information that you would put into a comment into the except message instead.